src/app/pipes/filter.pipe.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable, Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+    name: 'filter',
+})
+@Injectable()
+export class FilterPipe implements PipeTransform {
+    // tslint:disable-next-line: no-any
+    transform(items: any[], field: string, value: string, caseSensitive: boolean): any[] {
+        if (!items) { return []; }
+        if (!value || value.length === 0) { return items; }
+        try {
+            const reg = new RegExp(value, `${caseSensitive ? '' : 'i'}g`);
+            const results = items.filter(it => it[field] && value && it[field].match(reg));
+
+            return results.length > 0 ? results : [undefined];
```

Comment:
why `[undefined]` and not `[]`? Or just `results`?

Reply:
In the html template I need to show the message "No results":
```
<ng-container
    *cdkVirtualFor="let entity of list?.content | startsWith:'sortKey':(querySearch ? '' : selectedKey)| filter:'label':querySearch:caseSensitiveSearch; templateCacheSize: 0">
    <evt-named-entity *ngIf="entity" [inList]="true" [entity]="entity"> </evt-named-entity>
    <div class="p-3" *ngIf="!entity"> {{'noMatches' | translate }} </div>
</ng-container>
```
This was the only way I was able to achieve my goal. Do you have a different solution to suggest?

Reply:
Add `; let l = count` to the `ngFor` and change:
`*ngIf="entity"` => `*ngIf="count > 0"`
`*ngIf="!entity"` => `*ngIf="count <= 0"`